--- ocaml-data-notation-0.0.11/setup.ml.orig	2013-11-22 01:34:28.000000000 +0100
+++ ocaml-data-notation-0.0.11/setup.ml	2021-09-18 20:33:00.818590112 +0200
@@ -227,11 +227,9 @@ module OASISString = struct
       raise Not_found
 
   let replace_chars f s =
-    let buf = String.make (String.length s) 'X' in
-      for i = 0 to String.length s - 1 do
-        buf.[i] <- f s.[i]
-      done;
-      buf
+    let buf = Buffer.create (String.length s) in
+    String.iter (fun c -> Buffer.add_char buf (f c)) s;
+    Buffer.contents buf
 
 end
 
